internal/builders/docker/commands.go

```diff
@@ -70,11 +73,16 @@ func writeBuildDefinitionToFile(bd pkg.BuildDefinition, path string) error {
 func BuildCmd(check func(error)) *cobra.Command {
 	io := &pkg.InputOptions{}
 	var forceCheckout bool
+	var subjectsPath string
 
 	cmd := &cobra.Command{
 		Use:   "build [FLAGS]",
 		Short: "Builds the artifacts using the build config, source repo, and the builder image.",
 		Run: func(cmd *cobra.Command, args []string) {
+			// The BuildArtifacts function changes directory. Here we convert
+			// the output path `subjectsPath` to an absolute path.
+			outPath, err := filepath.Abs(subjectsPath)
```

Comment:
Can you make use of https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/utils/path.go#L40 and https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/utils/path.go#L74 to avoid directory traversal and overwrite of existing files?

/cc @asraa

Reply:
Ohh, maybe internally in `writeFile` so it can happen for buildDefinition output too.

Reply:
Thanks. Replaces it with `CreateNewFileUnderCurrentDirectory`. Also changed `writeToFile` to take a writer as input.